### PR TITLE
Refactor NFT classes to be interfaces, create util functions

### PR DESCRIPTION
**Pull request by deric-alchemy**

### CODE REVIEW

File: src/api/util.ts [hunk 1]
```diff
@@ -29,3 +32,55 @@ export function toHex(num: number): string {
 export function isHex(possibleHexString: string): boolean {
   return /^0x[0-9a-fA-F]+$/.test(possibleHexString);
 }
+
+export function getBaseNftFromRaw(
```
Comment (thebrianchen) [inline src/api/util.ts:36]: Can you move these to `src/util/util.ts`? I'm hoping to leave all methods in `src/api` as user-facing exports. 

Feel free to rename the other file to make it less confusing!

File: src/api/util.ts [hunk 2]
```diff
@@ -29,3 +32,55 @@ export function toHex(num: number): string {
...
+  rawBaseNft: RawBaseNft,
+  contractAddress: string
+): BaseNft {
+  return {
+    contract: { address: contractAddress },
+    tokenId: BigNumber.from(rawBaseNft.id.tokenId).toString(),
```
Comment (thebrianchen) [inline src/api/util.ts:42]: nit: can you move the comments that were originally on the class over to the util as well?

**Review by thebrianchen — APPROVED** [review]
lgtm! thanks for taking on the refactor

File: src/api/util.ts (hunk 1 above)
Comment (deric-alchemy) [reply src/api/util.ts:36]: done!

**Pull request merged**